### PR TITLE
Fixed sidebar in learn-ng

### PR DESCRIPTION
**Pull request by chinmaym07**
Signed-off-by: chinmaym07 <b418020@iiit-bh.ac.in>

**Description**

This PR fixes #

**Notes for Reviewers**
Made Sidebar Sticky for Chapter page & Slightly modified the styles of toc.

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

Issue Fixed :

https://user-images.githubusercontent.com/44670961/127337679-b8e2333e-6cee-467c-90dd-373ffbc666ad.mp4

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit c20c5ba750211c3c57edfbdaf2817ee3f829e736 at: https://610168398cb185118ff041f8--layer5ng.netlify.app

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Screenshot 2021-07-28 at 7 56 12 PM

That's looking great 🎉 
Can you look at this responsiveness issue ?

**Comment by l5io** [comment]
🚀 Preview for commit f6874432e3ecc9b36773bf4da3fe1ecedef771c3 at: https://6101738d342f2420da88ac18--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 0a2b4b4d89af5ce4b559d4d79218b04e19388f4d at: https://6102d2ffc325973b770ba004--layer5ng.netlify.app

File: src/sections/Learn-Layer5/Course-Overview/index.js [hunk 1]
```diff
@@ -68,7 +67,7 @@ const CourseOverview = ({ course, chapters }) => {
             {course.frontmatter.toc.map((item, index) => (
               <Link
                 key={index}
-                to={`./istio/${item}`}
+                to={`/learn-ng/${course.fields.learnpath}/${course.fields.course}/istio/${item}/`}
```
Comment (Nikhil-Ladha) [inline src/sections/Learn-Layer5/Course-Overview/index.js:70]: A relative path will work. Why change to use the absolute path?

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
The line connecting the dots on the sidebar is not properly aligned. Please fix that.

**Comment by l5io** [comment]
🚀 Preview for commit ffc006a5f0b51fef60e4197da616b429e4e53d2f at: https://61030edd4a903a866ef032fc--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 07ae502926e24d76827e84df4a706d45b15a5069 at: https://6103c99feb9ff5f71760b4ae--layer5ng.netlify.app

File: src/components/Learn-Components/TOC-Chapters/index.js [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+  const [path, setPath] = useState("");
+
   const reformatTOC= (data) => {
     let newData = data.split("-").join(" ");
     let firstLetter = newData.charAt(0).toUpperCase();
     newData = `${firstLetter}${newData.slice(1)}`;
     return newData;
   };
-
+
```
Comment (Nikhil-Ladha) [inline src/components/Learn-Components/TOC-Chapters/index.js:15]: nit: extra space

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
A good fix for the `active link` issue. `useEffect` is the exact solution we needed.
Though a small change is left i.e, the line connecting the dots is still not aligned properly.

**Comment by l5io** [comment]
🚀 Preview for commit 4e3ac7d1cbee436113bbefc492370af22afede47 at: https://6104090d2dce196b5c0f4e30--layer5ng.netlify.app

**Review by Nikhil-Ladha — APPROVED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 45375732189aa7a1804394c03ca32fa64ed6a22f at: https://6106cc87ade53d7fcb3d6b49--layer5ng.netlify.app

**Comment by leecalcote** [comment]
FYI
Screen Shot 2021-08-02 at 7 39 38 AM

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Screenshot 2021-08-02 at 6 12 24 PM

Can you also look at this line and dot alignment issue @chinmaym07 ?

**Comment by l5io** [comment]
🚀 Preview for commit a4c7774be298410c7f61b34646a135c7cd46e6f7 at: https://6108223fc714dafebce6f872--layer5ng.netlify.app

**Comment by sudo-NithishKarthik** [comment]
Screenshot 2021-08-03 at 10 34 43 AM

It is still not passing through the center, or, is it just in my laptop ? 😕

**Comment by chinmaym07** [comment]
I am not sure .. please tell me your resolution at which this issue occurs ?

image

**Comment by sudo-NithishKarthik** [comment]
> I am not sure .. please tell me your resolution at which this issue occurs ?
> 
> image

oh, that's weird. My screen resolution is 
**1440 x 900**

**Comment by chinmaym07** [comment]
Strange again 😀 

image

**Comment by sudo-NithishKarthik** [comment]
what browser are you using ?

**Comment by chinmaym07** [comment]
firefox

**Comment by sudo-NithishKarthik** [comment]
Ig its a browser specific issue, it's neither working with brave nor safari 😕 
can you check with these if possible ?

**Comment by Nikhil-Ladha** [comment]
It is a browser-specific issue for sure. It works fine for me in Firefox but is not aligned in brave and chrome.

**Comment by warunicorn19** [comment]
Also, @chinmaym07 there seems to be a merge conflict, if you could look into that.

**Comment by l5io** [comment]
🚀 Preview for commit f61387ef45f06437ee19b3c63bbac7772b5c5f56 at: https://6108f50616b098e1c26f75d4--layer5ng.netlify.app

**Comment by leecalcote** [comment]
Merge conflict fixed.

**Comment by l5io** [comment]
🚀 Preview for commit 1fb723cb908e95734460d6569b23d921eb1083a9 at: https://611123db842ddbc37b5fc589--layer5ng.netlify.app

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
The issues are fixed now 🎉 
Thank you @chinmaym07 
As much as i wan't to get this pr merged, there is just one small thing to be done 😄 , the service mesh switcher is not moving along with the toc, can you also make it sticky ?

**Comment by Nikhil-Ladha** [comment]
Tip: For the service mesh switcher to move along, move both the divs under a single div and make that div sticky, then it will work perfectly.

**Comment by chinmaym07** [comment]
I always mess up while solving conflicts .. sorry will open a new PR 😢

**Pull request closed without merging**